daq_main_common.py

```diff
@@ -29,10 +31,64 @@
 immediate_command_list = []
 z = 25
 
+def setGovState(state):
+  setGovRobot(gov_robot, state)
+
+
+functions = [anneal,
+  set_beamsize,
+  importSpreadsheet,
+  mvaDescriptor,
+  setTrans,
+  loop_center_xrec,
+  autoRasterLoop,
+  snakeRaster,
+  ispybLib.insertRasterResult,
+  backlightBrighter,
+  backlightDimmer,
+  changeImageCenterHighMag,
+  changeImageCenterLowMag,
+  center_on_click,
+  runDCQueue,
+  warmupGripper,
+  dryGripper,
+  enableDewarTscreen,
+  parkGripper,
+  stopDCQueue,
+  continue_data_collection,
+  mountSample,
+  unmountSample,
+  reprocessRaster,
+  fastDPNodes,
+  spotNodes,
+  unmountCold,
+  openPort,
+  set_beamcenter,
+  closePorts,
+  clearMountedSample,
+  recoverRobot,
+  rebootEMBL,
+  restartEMBL,
+  openGripper,
+  closeGripper,
+  homePins,
+  setSlit1X,
+  setSlit1Y,
+  testRobot,
+  setGovState]
+
+whitelisted_functions: "Dict[str, Callable]" = {
+    func.__name__: func for func in functions
+}
   
 def execute_command(command_s):
-  logger.info('executing command: %s' % command_s)
-  exec(command_s)
+  logger.info("executing command: %s" % command_s)
+  try:
```

Comment:
Consider making separate try/except blocks for _parsing_ the command (deserialize and function look-up) and _executing_ it (`func(...)`). The message `Error executing ....` could be misleading here.

Reply:
Addressed in [79071fe](https://github.com/NSLS-II/lsdc/pull/346/commits/79071fec2b4b13dd2640ddd57b3ac34f52bcaae7)